src/tasks/tasks.py

```diff
@@ -101,6 +101,45 @@ def _make_instance(input1, input2, labels, idx):
     #  return list(instances)
     return instances  # lazy iterator
 
+def create_subset_scorers(count, scorer_type, **args_to_scorer):
+    '''
+    Parameters:
+        count: N_tag, number of different "coarse__fine" tags
+        scorer_type: which scorer to use
+        **args_to_scorer: arguments passed to the scorer
+    Returns:
+        scorer_list: a list of N_tag scorer object
+    '''
+    scorer_list = [scorer_type(**args_to_scorer) for _ in range(count)]
+    return scorer_list
+
+def update_subset_scorers(scorer_list, estimations, labels, tagmask):
+    '''
+    Parameters:
+        scorer_list: a list of N_tag scorer object
+        estimations: a (bs, *) tensor, model estimation
+        labels: a (bs, *) tensor, ground truth
+        tagmask: a (bs, N_tag) 0-1 tensor, indicating tags of each sample
+    '''
+    for tid, scorer in enumerate(scorer_list):
+        subset_idx = torch.nonzero(tagmask[:, tid]).squeeze(dim=1)
+        subset_estimations = estimations[subset_idx]
+        subset_labels = labels[subset_idx]
+        if len(subset_idx) > 0:
+            scorer(subset_estimations, subset_labels)
+    return
```

Comment:
Do you need this?

Reply:
Yes, these functions can be used to easily implement fine grained evaluation by any task that needs it (like mnli-diagnostic), they are not restricted to any particular tag or score metric.